src/main/kotlin/org/rust/lang/core/psi/util/RustModExtensions.kt

```diff
@@ -77,13 +79,7 @@ internal val RustModItem.ownedDirectory: PsiDirectory?
     }
 
 private val RustModItem.isCrateRoot: Boolean
-    get() {
-        val file = containingFile.virtualFile ?: return false
-        val containingDir = file.parent
-        return project.getCrateSourceRootFor(file)?.let {
-            it.equals(containingDir) && (containingFile.name == RustModules.MAIN_RS || containingFile.name == RustModules.LIB_RS)
-        } ?: false
-    }
+    get() = getModule()?.crateRoots.orEmpty().contains(this)
```

Comment:
Bringing up PSI for all the `main.rs`/`lib.rs` files just to check whether this particular `mod` is the one of them seems fishy.